### PR TITLE
feat: Notification Subscriptions forV3

### PR DESCRIPTION
**Pull request by alexrisch**
Added notification subscribing for V3 only

## Summary by CodeRabbit

- **New Features**
	- Enhanced notification subscription logic with improved error handling and logging.
	- Updated conversation fetching process for better synchronization.

- **Bug Fixes**
	- Improved error handling during synchronization to ensure responsiveness to connection issues.

- **Refactor**
	- Simplified notification handling and conversation tracking within the chat store.
	- Updated method signatures and types for clarity and maintainability.

- **Chores**
	- Removed legacy code and unnecessary subscriptions to streamline functionality.

### CODE REVIEW

File: utils/notifications.ts [hunk 1]
```diff
@@ -40,161 +46,98 @@ export const lastNotifSubscribeByAccount: {
...
+  try {
+    logger.info(
+      "[subscribeToNotifications] start",
+      account,
+      conversations.length
+    );
+    const thirtyDayPeriodsSinceEpoch = Math.floor(
+      Date.now() / 1000 / 60 / 60 / 24 / 30
+    );
+    const client = (await getXmtpClient(account)) as ConverseXmtpClientType;
```
Comment (thierryskoda) [inline utils/notifications.ts:67]: I see we do A LOT of those `const client = (await getXmtpClient(account)) as ConverseXmtpClientType;` could we make it simpler and at least not need the `as ... ` just curious

File: utils/notifications.ts [hunk 2]
```diff
@@ -40,161 +46,98 @@ export const lastNotifSubscribeByAccount: {
...
+    };
+
+    logger.info("[subscribeToNotifications] saving notifications subscribe");
+    await saveNotificationsSubscribe(
+      account,
+      nativePushToken,
+      nativeTokenQuery.type,
+      nativeTokenQuery.type === "android" ? "converse-notifications" : null,
+      topicsToUpdateForPeriod
+    );
```
Comment (thierryskoda) [inline utils/notifications.ts:133]: Many params , would love a object instead 👀

File: utils/xmtpRN/sync.ts [hunk 3]
```diff
@@ -146,8 +146,8 @@ const syncClientConversationList = async (account: string) => {
     });
     // Prefetch the conversation list so when we land on the conversation list
     // we have it ready, this will include syncing all groups
-    const conversationList = await prefetchConversationListQuery(account);
-    // TODO: Update notification topics
+    const conversationList = await fetchConversationListQuery(account);
+    subscribeToNotifications({ conversations: conversationList, account });
   } catch (e) {
     logger.error(e, {
       context: `Failed to fetch persisted conversation list for ${account}`,
```
Comment (thierryskoda) [inline utils/xmtpRN/sync.ts:153]: would be cool to start handling errors better. Maybe worth waiting until we have a clear strategy tho

File: utils/notifications.ts (hunk 1 above)
Comment (alexrisch) [reply utils/notifications.ts:67]: Yeah I agree we don't need it, it's there because of old web stuff, so we should probably remove it

**Pull request merged**